.github/workflows/docker.yml

```diff
@@ -1,23 +1,21 @@
 name: Docker
+
 on:
   pull_request:
-    paths:
-      - .github/workflows/docker.yml
-      - Dockerfile
   push:
-    paths:
-      - .github/workflows/docker.yml
-      - Dockerfile
-    branches-ignore:
+    branches:
       - master
+  merge_group:
   release:
     types:
       - published
```

Comment:
I wonder if there's a way to reuse the built Docker image from the `push` job so we don't need to rebuild from scratch when a release has been cut. (Or does Docker already do that? No idea.)

Reply:
Docker does not. Technically we can do this if we publish a Git SHA-tagged image on `master` updates so that we can retrieve it and version-tag it. We currently do not, and we only publish the `homebrew/ubuntu22.04:master` image (i.e., not for 18.04, 20.04, 24.04).

Reply:
We can do some caching if we use https://github.com/docker/build-push-action